tools/deployment/k8s/s3/deploy.sh

```
@@ -0,0 +1,136 @@
+    local message=$2
+    echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") [$verbosity] $message" >&2
+}
+
+# Function to wait until S3 endpoint is available by listing the available buckets from the S3 endpoint.
+# If the listing operation succeeds, the function returns immediately.
+# Otherwise, it retries a maximum of 10 times with a delay of 6 seconds between each retry.
+wait_for_s3_availability() {
+    local MAX_RETRIES=10
+    local RETRY_DELAY_IN_SECS=6
+
+    for ((retries = 0; retries < $MAX_RETRIES; retries++)); do
```

Shellcheck (0.10.0)

[style] $/${} is unnecessary on arithmetic variables.

(SC2004)